var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -131,6 +153,14 @@ def cmake_args(self):
             '-DXERCESC_ROOT_DIR={0}'.format(spec['xerces-c'].prefix)
         ]
 
+        # Use the correct C++ standard option for the requested version
+        if spec.version >= Version('11.0'):
+            options.append(
+                self.define_from_variant('CMAKE_CXX_STANDARD', 'cxxstd'))
+        else:
+            options.append(
+                self.define_from_variant('GEANT4_BUILD_CXXSTD', 'cxxstd'))
```

Comment:
Possible alternative:
```suggestion
       options.append(self.define_from_variant(
           'CMAKE_CXX_STANDARD' if spec.version >= Version('11.0') else 'GEANT4_BUILD_CXXSTD',
           'cxxstd'
       ))
```